### ISSUE TITLE
Cortx-s3server doesn't accept certain multipart uploads

### ISSUE THREAD

**Issue by adgang** [Seagate/cortx-s3server#876]
s3 server requires the first part of multipart upload to have a size value as integer multiple of unit_size.

https://github.com/Seagate/cortx-s3server/blob/8428644d6ac55803b9f1da4916bfc61835ecb06d/server/s3_put_multiobject_action.cc#L361

This causes problems or at least surprises while using Cortx as storage backend. 

If this restriction is intended for something meaningful like performance, it has to be at least documented.

Example error(encountered while [integrating Pulsar](https://github.com/Seagate/cortx/pull/969)):
```
error after writing 65626/3283684 bytes to http://s3.seagate.com/pulsar-topic-test/70cec628-d533-4b0e-b1fa-da3cb132cfc6-ledger-121?partNumber=1&uploadId=cfe51210-144e-4274-953b-5cf2ab77c595
...
request PUT http://s3.seagate.com/pulsar-topic-test/9f8ec818-a0f6-41fb-bdf1-46c6fa7828d8-ledger-148?partNumber=1&uploadId=86d20521-741d-4b54-ac5e-6771b8f84bd3 HTTP/1.1 failed with code 400, error: AWSError{requestId='e8684e41-2e56-4d65-8835-0a36c4013cb4', requestToken='null', code='InvalidPartSize', message='Part size must be multiple of unit_size. Refer s3 readme.', context='{Resource=pulsar-topic-test/9f8ec818-a0f6-41fb-bdf1-46c6fa7828d8-ledger-148}'}
```

To reproduce the issue, 

1. Create a file large-test-file.001 with x bytes of content. Example x=8, x just has to be a non-integer multiple of motr unit_size(which is 1MB by default).
2. Start a multi part upload and get upload id from s3 server:
```
$ aws s3api create-multipart-upload --bucket pulsar-topic-test --key large-test-file
example-bucket	large-test-file	d3a5fbaf-69b0-44b0-96eb-cce37c47d42a
```
3. Upload the first part against the upload id:
```
$ aws  s3api upload-part --bucket example-bucket --key large-test-file --part-number 1 --body large-test-file.001 --upload-id d3a5fbaf-69b0-44b0-96eb-cce37c47d42a --content-md5 exampleaAmjr+4sRXUwf0w==

An error occurred (InvalidPartSize) when calling the UploadPart operation: Part size must be multiple of unit_size. Refer s3 readme.
```

**Comment by nileshgovande** [comment]
Tagging @DPG17 - 
@adgang There are limitations

**Comment by adgang** [comment]
Thanks for the response @nileshgovande. I understand that. The [compatibility matrix](https://github.com/Seagate/cortx-s3server/blob/8428644d6ac55803b9f1da4916bfc61835ecb06d/docs/s3-supported-api.md) hints at it. IMO there is satisfactory level of support for multipart upload, even if the above issue is not fixed as bug(the error message was informative as well). It's just that if documentation reflects what limitations should be expected specifically, and how they can be worked around, there will not be surprises in DX.

A couple of workarounds to mitigate issue, if others are facing similar problem, are:
1. Tune your part size in multipart s3 client to a value higher than 1MB and an integer multiple of MBs. And if that is not possible(if a client library does not allow it beyond a part size, say), use the solution below.
2. Tune S3_MOTR_LAYOUT_ID in `conf/s3config.yaml` to lower values and restart cortx.
3. Do not use multi-part uploads for smaller file sizes(smallest unit_size is 1KB so for smaller sizes, even 2. above won't help). This needs some conditional code path and developer effort.

These are fairly simple measures for a developer to follow to get Cortx s3 working for what they want seamlessly(at least I did not find any other issues myself).

**Comment by t7ko-seagate** [comment]
This limitation should have been addressed (removed) with recent merge of Multipart Redesign, which is available in commit c50dac13 and later.  We no longer force any specific limits on part sizes.  I'll be closing this issue -- please reopen if some further issues found.
